resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -313,7 +313,7 @@ private[spark] class ApplicationMaster(
           sparkConf.get(DRIVER_PORT)),
         YarnSchedulerBackend.ENDPOINT_NAME)
       // The client-mode AM doesn't listen for incoming connections, so report an invalid port.
-      registerAM(Utils.localHostName, -1, sparkConf,
+      registerAM(Utils.localHostNameForURI(), -1, sparkConf,
```

Comment:
`localHostName` not work for ipv6? I saw its code comment `In case of IPv6, getHostAddress may return '0:0:0:0:0:0:0:1'.`.

Reply:
For URI, we need to wrap `[]` to distinguish IP and port. `Utils.localHostNameForURI` do that additionally.